src/AgentHeaderDescriptor.php

```diff
@@ -43,44 +41,87 @@ class AgentHeaderDescriptor
     // TODO(michaelbausor): include bumping this version in a streamlined
     // release process. Issue: https://github.com/googleapis/gax-php/issues/48
     const GAX_VERSION = '0.7.0';
+    const UNKNOWN_HEADER = '';
 
-    private $clientName;
-    private $clientVersion;
-    private $codeGenName;
-    private $codeGenVersion;
-    private $gaxVersion;
-    private $phpVersion;
+    private $metricsHeaders;
 
     /**
      * @param array $headerInfo {
-     *     Required.
+     *     Optional.
      *
-     *     @type string $clientName the name of the client application.
-     *     @type string $clientVersion the version of the client application.
+     *     @type string $phpVersion the PHP version.
+     *     @type string $libName the name of the client application.
+     *     @type string $libVersion the version of the client application.
      *     @type string $codeGenName the code generator name of the client library.
      *     @type string $codeGenVersion the code generator version of the client library.
      *     @type string $gaxVersion the GAX version.
-     *     @type string $phpVersion the PHP version.
+     *     @type string $grpcVersion the gRPC version.
+     *     @type string $additionalMetrics a dictionary of additional metrics for the header.
```

Comment:
Scratch `$additionalMetrics`. The service control team wants tight control over what we pass. The only thing we should leave somewhat open is `$libName`.